pkg/services/ngalert/store/alertmanager.go

```diff
@@ -13,22 +13,73 @@ var (
 	ErrNoAlertmanagerConfiguration = fmt.Errorf("could not find an Alertmanager configuration")
 )
 
+func getAlertmanagerConfigurationByID(sess *sqlstore.DBSession, id int64) (*models.AlertConfiguration, error) {
+	c := &models.AlertConfiguration{}
+
+	has, err := sess.ID(id).Get(c)
+	if !has {
+		return nil, ErrNoAlertmanagerConfiguration
+	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Checking the non-error return value first can be buggy, since if there's an error there won't be any obligation to return anything sensible for non-error return values.

```suggestion
	if err != nil {
		return nil, err
	}
	if !has {
		return nil, ErrNoAlertmanagerConfiguration
	}
```